src/server/apsystem.ts

```diff
@@ -192,7 +221,86 @@ export default class ActivityPubSystem {
     })
   }
 
-  async sendTo (actorURL: string, fromActor: string, activity: APActivity): Promise<void> {
+  async get (fromActor: string, urlOrObject: ObjectField): Promise<APObject> {
+    if (typeof urlOrObject === 'string') {
+      const response = await this.signedFetch(fromActor, {
+        method: 'GET',
+        url: urlOrObject,
+        headers: {}
+      })
+      if (!response.ok) {
+        throw createError(
+          500,
+          `Cannot fetch data for ${fromActor} at ${urlOrObject}: http status ${response.status} - ${await response.text()}`
+        )
+      }
+
+      return await response.json()
+    } else {
+      return urlOrObject
+    }
+  }
+
+  async * getAll (
+    fromActor: string,
+    items: ObjectField[]
+  ): AsyncIterableIterator<APObject> {
+    for (const itemOrUrl of items) {
+      const item = await this.get(fromActor, itemOrUrl)
```

Comment:
won't this fail for the first get() that throws?

Reply:
Yes, I assume if you have outbox item errors we should stop trying to iterate through it as soon as we face one. It'd happen if you had invalid activities with IDs that can't be resolved